google/resource_kms_crypto_key_test.go

```diff
@@ -0,0 +1,381 @@
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testGoogleKmsCryptoKey_rotation(projectId, projectOrg, projectBillingAccount, keyRingName, cryptoKeyName, rotationPeriod),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleKmsCryptoKeyExists("google_kms_crypto_key.crypto_key"),
+					testAccCheckGoogleKmsCryptoKeyHasRotationParams(rotationPeriod, "google_kms_crypto_key.crypto_key"),
+				),
+			},
+			resource.TestStep{
+				Config: testGoogleKmsCryptoKey_removed(projectId, projectOrg, projectBillingAccount, keyRingName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGoogleKmsCryptoKeyWasRemovedFromState("google_kms_crypto_key.crypto_key"),
+					testAccCheckGoogleKmsCryptoKeyVersionsDestroyed(projectId, location, keyRingName, cryptoKeyName),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckGoogleKmsCryptoKeyExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Resource not found: %s", resourceName)
+		}
+
+		keyRingId, err := parseKmsKeyRingId(rs.Primary.Attributes["key_ring"], config)
+
+		if err != nil {
+			return err
+		}
+
+		cryptoKeyId := &kmsCryptoKeyId{
+			KeyRingId: *keyRingId,
+			Name:      rs.Primary.Attributes["name"],
+		}
+
+		listCryptoKeysResponse, err := config.clientKms.Projects.Locations.KeyRings.CryptoKeys.List(cryptoKeyId.parentId()).Do()
+		if err != nil {
+			return fmt.Errorf("Error listing KeyRings: %s", err)
+		}
+
+		for _, cryptoKey := range listCryptoKeysResponse.CryptoKeys {
+			log.Printf("[DEBUG] Found CryptoKey: %s", cryptoKey.Name)
+
+			if cryptoKey.Name == cryptoKeyId.cryptoKeyId() {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("CryptoKey not found: %s", cryptoKeyId.cryptoKeyId())
+	}
+}
+
+func testAccCheckGoogleKmsCryptoKeyHasRotationParams(rotationPeriod, resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		config := testAccProvider.Meta().(*Config)
+
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Resource not found: %s", resourceName)
+		}
+
+		keyRingId, err := parseKmsKeyRingId(rs.Primary.Attributes["key_ring"], config)
+
+		if err != nil {
+			return err
+		}
+
+		cryptoKeyId := &kmsCryptoKeyId{
+			KeyRingId: *keyRingId,
+			Name:      rs.Primary.Attributes["name"],
+		}
+
+		getCryptoKeyResponse, err := config.clientKms.Projects.Locations.KeyRings.CryptoKeys.Get(cryptoKeyId.cryptoKeyId()).Do()
+
+		if err != nil {
+			return err
+		}
+
+		if rotationPeriod != getCryptoKeyResponse.RotationPeriod {
+			return fmt.Errorf("Expected rotation period %s to match input %s", getCryptoKeyResponse.RotationPeriod, rotationPeriod)
+		}
+
+		_, err = time.Parse(time.RFC3339Nano, getCryptoKeyResponse.NextRotationTime)
+
+		return err
```

Comment:
It might be worth it to wrap this error, something like `return fmt.Errorf("Error parsing NextRotationTime: %s", err)`